### ISSUE TITLE
When sharing a large file with --local-only, the download quickly stalls

### ISSUE THREAD

**Issue by micahflee** [onionshare/onionshare#870]
While implementing #868 I found a very strange, and reproducible, bug. I'm sharing a 352mb file, `large_file.img`, I had laying around, and I start sharing it like this:

```
./dev_scripts/onionshare --local-only --debug ~/tmp/large_file.img
```

Since it's `--local-only`, I get a URL like http://127.0.0.1:17615/overtone-risk. When I start downloading it in a browser, it occasionally works, but most of the time it fails just after 3% of the download. Only instead of failing, it just stalls.

After some debugging, I think it's happening in `onionshare/web/share_mode.py`, in `ShareModeWeb.define_routes()`, in `download_logic()`, in `generate()`. The generator is supposed to load a file, read a chunk of it at a time, and yield a chunk. But it looks like it's stalling on this line:

```python
yield chunk
```

It simply never runs the following lines of code, and in the browser the download is active but says "Unknown time left -- 0 bytes (0 bytes/sec)".

I can reproduce nearly every time (but occasionally it works and downloads all the way).

When I do this same thing, but without `--local-only` and when using Tor Browser, it always works, and I can't reproduce the problem.

**Comment by micahflee** [comment]
Hey @farhaanbukhsh I'd be happy for you to contribute and fix this bug! You can go ahead and try to understand what's broken and post comments here, and make a pull request when you think you have a solution.

**Comment by mig5** [comment]
I'm not reproducing this lately. I'm able to download a 2.2GB file via `--local` mode (with the fix I'm working on for #1095)

**Comment by mig5** [comment]
Do you still reproduce this @micahflee ? It continues to work ok for me, sharing a 1.5GB file to myself via `--local-only`.
